src/entity-manager/EntityManager.ts

```diff
@@ -538,6 +539,35 @@ export class EntityManager {
         return qb.getOne();
     }
 
+    /**
+     * Finds first entity that matches given find options or rejects the returned promise on error.
+     */
+    findOneOrFail<Entity>(entityClass: ObjectType<Entity>|string, id?: string|number|Date|ObjectID, options?: FindOneOptions<Entity>): Promise<Entity>;
+
+    /**
+     * Finds first entity that matches given find options or rejects the returned promise on error.
+     */
+    findOneOrFail<Entity>(entityClass: ObjectType<Entity>|string, options?: FindOneOptions<Entity>): Promise<Entity>;
+
+    /**
+     * Finds first entity that matches given conditions or rejects the returned promise on error.
+     */
+    findOneOrFail<Entity>(entityClass: ObjectType<Entity>|string, conditions?: DeepPartial<Entity>, options?: FindOneOptions<Entity>): Promise<Entity>;
+
+    /**
+     * Finds first entity that matches given conditions or rejects the returned promise on error.
+     */
+    findOneOrFail<Entity>(entityClass: ObjectType<Entity>|string, idOrOptionsOrConditions?: string|string[]|number|number[]|Date|Date[]|ObjectID|ObjectID[]|FindOneOptions<Entity>|DeepPartial<Entity>, maybeOptions?: FindOneOptions<Entity>): Promise<Entity> {
+        return this.findOne(entityClass, idOrOptionsOrConditions as any, maybeOptions).then((value) => {
+            if (value === undefined) {
+                return Promise.reject(new EntityNotFoundError(
+                        (typeof entityClass === "string") ? entityClass : entityClass.constructor.name,
+                        JSON.stringify(idOrOptionsOrConditions)));
```

Comment:
I have to stringify this parameter otherwise the error message could show a frustrating `[Object object]`. Can you make sure this is safe? If there is a circular object here we end up throwing an error while attempting to create another error :/

Reply:
> Can you make sure this is safe?

Not really sure... We can either wrap into try/catch or show `idOrOptionsOrConditions` only if its a primitive.

btw move all boilerplate code into `EntityNotFoundError` to make this code cleaner.